File: my-folder/0209-minimum-size-subarray-sum/solution.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        
        if (nums.size()==1)
        {
            if (target<nums[0])
            {
                return 1;
            }
            else
            {
                return 0;
            }
        }

        if (nums[0]>target)
        {
            return 1;
        }
        //sort(nums.begin(),nums.end());
        int i = 0;
        int  j = 1;
        vector<int> output;
        int sum = accumulate(nums.begin(),nums.end(),0);
        int loop_sum = nums[i]+nums[j];
        if (sum>=target)
        {
            while(j<nums.size())
            {
                

                if (loop_sum<target)
                {
                    j++;
                    if (j<nums.size())
                    {
                        loop_sum +=  (nums[j]); 
                    }
                }
                else
                {
                    output.push_back((j-i)+1);
                    loop_sum = loop_sum -nums[i];
                    i++;
                    
                }

            }
            
            auto min = min_element(output.begin(),output.end());
            return *min;

        }
        else
        {
            return 0;
        }


        
    }
};

File: my-folder/0209-minimum-size-subarray-sum/solution.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n = len(nums)
        min_len = float('inf')

        left = 0
        right = 0

        running_total = 0

        while right < n:
            running_total+=nums[right]

            while left<=right and running_total >= target:
                min_len = min(min_len, (right-left)+1)
                running_total-=nums[left]

                left+=1
            right+=1
        
        return min_len if min_len != float('inf') else 0

                


